controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -108,23 +108,22 @@ void dynamic_channel_selection_task::work()
             LOG(DEBUG) << "single scan is pending, trigger single scan";
             fsm_move_state(eState::TRIGGER_SCAN);
         } else if (database.get_channel_scan_is_enabled(m_radio_mac)) {
-            auto now = std::chrono::steady_clock::now();
-
-            if (now > m_next_scan_timestamp_interval) {
+            if (std::chrono::steady_clock::now() > m_next_scan_timestamp_interval) {
```

Comment:
I still think that there's no sense to move into the `TRIGGER_SCAN` state if you can't really trigger a scan.
The task should remain in `IDLE` until it's able to trigger a scan.

Reply:
please see the last fixup - it addresses both.
I will edit the commit message accordingly as part of the squash